tools/unique_regions/unique_regions.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python
+
+import os
+import argparse
+
+def remove_first_line(s):
+    return s[s.find('\n')+1:s.rfind('\n')]
+
+def build_cont(prev1, prev2, prev3):
+    return "%s\t%i\t%i\n" % (prev1, prev2, prev3)
+
+def overlap(infile):
+    overlapping = ""
+    unique = ""
+    prev_chromo = ""
+    prev_start = 0
+    prev_end = 0
+
+    #iterating the bed file, line by line
+    for line in open(infile, 'r'):
+        split_line = line.split("\t")
+        chromo = str(split_line[0])
+        start = int(split_line[1])
+        end = int(split_line[2])
+
+        #verifying whether the two successive regions are on the same chromosome,
+        #if not, the comparison will be skipped
+        if chromo != prev_chromo:
+            prev_chromo = chromo
+            unique += build_cont(prev_chromo, prev_start, prev_end)
+        else:
+            #seperating the unique regions
+            if start >= prev_end:
+                unique += build_cont(prev_chromo, prev_start, prev_end)
+            else:
+                #seperating the overlapping regions
+                overlapping += build_cont(prev_chromo, prev_start, prev_end)
+        prev_end = end
+        prev_start = start
+
+    #writing the last region
+    unique += build_cont(prev_chromo, prev_start, prev_end)
+    unique = remove_first_line(unique)
+    return unique, overlapping
+
+def unique_regions(args):
+    l = 0
+    #retrieving the two output groups (unique and overlapping regions)
+    unique, overlapping = overlap(args.input)
```

Comment:
it can be integrated in the while loop as follows:
    while l == 0 or overlapping != "":
        if l == 0:
            args.input = XXXX
        else:
            args.input = "temp_overlapping_regions.bed"
How would I replace XXXX to read from the original file??